nucypher/config/node.py

```diff
@@ -623,62 +629,61 @@ def initialize(self, password: str, download_registry: bool = True) -> str:
                 raise self.ConfigurationError("There is already a keyring attached to this configuration.")
             return
 
-        if (checksum_address or self.checksum_public_address) is None:
+        if (checksum_address or self.checksum_address) is None:
             raise self.ConfigurationError("No account specified to unlock keyring")
 
         self.keyring = NucypherKeyring(keyring_root=self.keyring_dir,  # type: str
-                                       account=checksum_address or self.checksum_public_address,  # type: str
+                                       account=checksum_address or self.checksum_address,  # type: str
                                        *args, **kwargs)
 
-    def write_keyring(self, password: str, **generation_kwargs) -> NucypherKeyring:
+    def write_keyring(self, password: str, wallet: bool = True, **generation_kwargs) -> NucypherKeyring:
+
+        checksum_address = None
 
         #
         # Decentralized
         #
+        if wallet:
 
-        # Note: It is assumed the blockchain is not yet available.
-        if not self.federated_only and not self.checksum_public_address:
-
-            # "Casual Geth"
-            if self.provider_process:
+            # Note: It is assumed the blockchain is not yet available.
+            if not self.federated_only and not self.checksum_address:
 
-                if not os.path.exists(self.provider_process.data_dir):
-                    os.mkdir(self.provider_process.data_dir)
+                # "Casual Geth"
+                if self.provider_process:
 
-                # Get or create wallet address (geth etherbase)
-                checksum_address = self.provider_process.ensure_account_exists(password=password)
+                    if not os.path.exists(self.provider_process.data_dir):
+                        os.mkdir(self.provider_process.data_dir)
 
-            # "Formal Geth" - Manual Web3 Provider, We assume is already running and available
-            else:
-                self.connect_to_blockchain()
-                if not self.blockchain.interface.w3.eth.accounts:
-                    raise self.ConfigurationError(f'Web3 provider "{self.provider_uri}" does not have any accounts')
-                checksum_address = self.blockchain.interface.w3.eth.accounts[0]  # TODO: Make this a configurable default in config files
+                    # Get or create wallet address (geth etherbase)
+                    checksum_address = self.provider_process.ensure_account_exists(password=password)
 
-            # Addresses read from some node keyrings (clients) are *not* returned in checksum format.
-            checksum_address = to_checksum_address(checksum_address)
+                # "Formal Geth" - Manual Web3 Provider, We assume is already running and available
+                else:
+                    self.connect_to_blockchain()
+                    if not self.blockchain.interface.w3.eth.accounts:
```

Comment:
I believe this was added to a function in the `Web3Client`. If not, we should add the function.

Reply:
:heavy_check_mark: